### PR TITLE
Ability to change notch behavior per block

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Mostly looking for feedback to start.

BlockSvg has a whole bunch of constants. Some are only used internally, some are exposed externally. Some of them depend on each other.

As part of an upcoming feature, I want to be able to have different blocks display themselves differently. To start with, I want to display the notch differently, but ultimately also want the shape to be different. Changing the notch just requires modifying a couple of our constants, but because these are currently shared, there's no way to have two different blocks.

This PR moves some of these constants onto the BlockSvg object, which means a future commit could change them for particular blocks.

I'm looking on feedback on two things in particular
(1) How I handle innerTopLeftCorner. I could have this remain a string rather than a function, and just depend on updating the string whenever notchPathRight changes, but this felt cleaner to me. Thoughts?
(2) Connection highlighting
- As written right now, I'm accessing block's svg_, which is a private
- Really it feels like BlockSvg should own generating the path for highlighting the connection. I started to do this, but it still didn't feel super clean for a couple reasons
  - Requires a pass through function on Block (so that we're not directly accessing the private svg_)
  - Connection wants to know about the generated DOM element. We could have the BlockSvg.highlightConnection return the element, which might be fine but feels a little unnatural.
- thoughts?

### CODE REVIEW

File: blockly-core/core/block_svg.js [hunk 1]
```diff
@@ -34,6 +41,21 @@ goog.require('goog.userAgent');
...
+   */
+  this.notchPathLeft = 'l 6,4 3,0 6,-4';
+  /**
+   * SVG path for drawing next/previous notch from left to right with
+   * highlighting.
+   */
+  this.notchPathLeftHighlight = 'l 6.5,4 2,0 6.5,-4';
+  /**
+   * SVG path for drawing next/previous notch from right to left.
+   * @const
```
Comment (bcjordan) [inline blockly-core/core/block_svg.js]: Should others be `@const` as well?

Comment (Bjvanminnen) [inline blockly-core/core/block_svg.js]: None of these should be const anymore. Will remove

**Comment by bcjordan** [comment]
> Connection wants to know about the generated DOM element. We could have the BlockSvg.highlightConnection return the element, which might be fine but feels a little unnatural.

Is that this?

```
Blockly.Connection.prototype.unhighlight = function() {
  goog.dom.removeNode(Blockly.Connection.highlightedPath_);
  delete Blockly.Connection.highlightedPath_;
};
```

Hmm. Couple of potential options:

1.

have `Blockly.Connection.prototype.highlight` call `this.sourceBlock_.highlightConnectionBelow()` and have `Blockly.Connection.prototype.unhighlight` call `this.sourceBlock_.unHighlightConnectionBelow()`, which perhaps in turn lets `blockSvg` create and destroy the highlight (or wedge a new class `ConnectionHighlightSvg` somewhere?)

2. 

Same as 1, but have blockSvg snag the blockSvg reference, e.g.

`this.sourceBlock_.getRenderer().highlightConnectionBelow()` and `this.sourceBlock_.getRenderer().unHighlightConnectionBelow()`

**Comment by Bjvanminnen** [comment]
Ended up going with the most simple fix of explicitly allowing classes like Connection to get BlockSvg from the Block.

**Comment by islemaster** [comment]
Seems sensible to gradually move this ownership to BlockSvg.  I like it.

**Pull request merged**